### PR TITLE
bundlewrap: init at 4.15.0

### PR DESCRIPTION
**Pull request by wamserma**
###### Description of changes

Add https://bundlewrap.org/, a (semi) declarative config-management tool with a low learning curve.
I've been maintaining it in my personal NUR for some years and use it with a small Debian VPS.
The tool is now mature enough (e.g. update frequency is low enough) to move this to nixpkgs.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [Y] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
> The tool is now mature enough (e.g. update frequency is low enough) to move this to nixpkgs.

FYI That is not a criteria for a package in nixpkgs.

File: pkgs/development/python-modules/bundlewrap/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+buildPythonPackage rec {
+  pname = "bundlewrap";
+  version = "4.15.0";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "bundlewrap";
+    repo = "bundlewrap";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/bundlewrap/default.nix:27]:
```suggestion
    rev = version;
```

File: pkgs/development/python-modules/bundlewrap/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    sha256 = "sha256-O31lh43VyaFnd/IUkx44wsgxkWubZKzjsKXzHwcGox0";
+  };
+
+  propagatedBuildInputs = [
+    cryptography jinja2 Mako passlib pyyaml requests setuptools tomlkit librouteros
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/bundlewrap/default.nix:32]: setuptools goes to nativeBuildInputs.

File: pkgs/development/python-modules/bundlewrap/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  ] ++ lib.optional (pythonOlder "3.11") [ rtoml ];
+
+  pythonImportsCheck = [ pname ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/bundlewrap/default.nix:35]:
```suggestion
  pythonImportsCheck = [ "setuptools" ];
```

File: pkgs/development/python-modules/bundlewrap/default.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  checkInputs = [ pytest ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/bundlewrap/default.nix:37]:
```suggestion
  checkInputs = [ pytestCheckHook ];
```

File: pkgs/development/python-modules/bundlewrap/default.nix [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    # only unit tests as integration tests need a OpenSSH client/server setup
+    py.test tests/unit
+  '';
+
+  meta = with lib; {
+    homepage = "https://bundlewrap.org/";
+    description = "Easy, Concise and Decentralized Config management with Python";
+    license = [ licenses.gpl3 ] ;
+    maintainers = with maintainers; [ wamserma ];
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/bundlewrap/default.nix:49]:
```suggestion
```
that is the default from buildPythonPackage

File: pkgs/top-level/python-packages.nix [hunk 6]
```diff
@@ -1583,6 +1583,8 @@ in {
 
   bunch = callPackage ../development/python-modules/bunch { };
 
+  bundlewrap = callPackage ../development/python-modules/bundlewrap { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:1586]: If this is only a cli tool we do not need or want this entry.

File: pkgs/development/python-modules/bundlewrap/default.nix [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  ] ++ lib.optional (pythonOlder "3.11") [ rtoml ];
+
+  pythonImportsCheck = [ pname ];
+
+  checkInputs = [ pytest ];
+
+  checkPhase = ''
+    # only unit tests as integration tests need a OpenSSH client/server setup
+    py.test tests/unit
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/bundlewrap/default.nix:42]:
```suggestion
  pytestFlagsArray = [
    # only unit tests as integration tests need a OpenSSH client/server setup
    "tests/unit"
  ];
```

**Comment by wamserma** [comment]
> > The tool is now mature enough (e.g. update frequency is low enough) to move this to nixpkgs.
> 
> FYI That is not a criteria for a package in nixpkgs.

I know, but frequent updates are easier to manage in NUR. :)

File: pkgs/top-level/python-packages.nix (hunk 6 above)
Comment (wamserma) [reply pkgs/top-level/python-packages.nix:1586]: It is mainly intended for CLI use, but I can imagine people bypassing the CLI and directly hooking into bundlewrap from their CD scripts.

**Pull request merged**